MahApps.Metro/Styles/Controls.PasswordBox.xaml

```diff
@@ -9,6 +10,198 @@
+                            <TextBlock x:Name="TextBoxHint"
+									    Text="{TemplateBinding Controls:TextBoxHelper.Watermark}"
+					                    Grid.Column="0" Grid.Row="0"
+                                        FontSize="15"
+                                       Padding="6,0,6,0"
+                                       Margin="2,0,2,0"
+                                        IsHitTestVisible="False"
+                                        Opacity="0.6"
+										Cursor="IBeam" Foreground="{TemplateBinding Foreground}"
+                                       HorizontalAlignment="Stretch"
+                                       VerticalAlignment="Center"
+										Visibility="{Binding ElementName=RevealedPassword, Path=Text, Converter={StaticResource StringToOppositeVisibilityConverter}}" />
+                            <Button x:Name="RevealButton"
+									Grid.Column="1" Grid.Row="0" SnapsToDevicePixels="True"
+									Style="{StaticResource RevealButtonStyle}"
+									Visibility="{Binding ElementName=RevealedPassword, Path=Text, Converter={StaticResource StringToVisibilityConverter}}">
+                            </Button>
+                        </Grid>
+                        <Rectangle x:Name="DisabledVisualElement"
+                                Stroke="{DynamicResource ControlsDisabledBrush}"
+                                StrokeThickness="{TemplateBinding BorderThickness, Converter={StaticResource ThicknessToDoubleConverter}}"
+                                Fill="{DynamicResource ControlsDisabledBrush}"
+                                IsHitTestVisible="False"
+                                Opacity="0" />
+                    </Grid>
+                    <ControlTemplate.Triggers>
+                        <Trigger Property="IsEnabled" Value="false">
+                            <Setter Property="Foreground" Value="{StaticResource {x:Static SystemColors.GrayTextBrushKey}}" />
+                            <Setter Property="Opacity" TargetName="RevealButton" Value="0.5" />
+                            <Setter Property="Text" TargetName="TextBoxHint" Value="{}{disabled} "/>
+                            <Setter TargetName="DisabledVisualElement" Property="Opacity" Value="0.6" />
+                        </Trigger>
+                        <DataTrigger Binding="{Binding ElementName=RevealButton, Path=IsPressed}" Value="True">
+                            <Setter TargetName="RevealedPassword" Property="Visibility" Value="Visible" />
+                            <Setter Property="Foreground" Value="Transparent" />
+                            <Setter Property="PasswordChar" Value=" " />
+
+
+                        </DataTrigger>
+                        <DataTrigger Binding="{Binding RelativeSource={RelativeSource Self}, Path=Text}" Value="">
+                            <Setter TargetName="TextBoxHint" Property="Visibility" Value="Visible" />
+                        </DataTrigger>
```

Comment:
this `DataTrigger` is not necessary and wrong (there is no `Text` property on `Password` box)